### ISSUE TITLE
sparsemax returns values that do not sum to 1 on all backends

### ISSUE THREAD

**Issue by SEPURI-SAI-KRISHNA** [keras-team/keras#23425]
`keras.ops.sparsemax` and `keras.activations.sparsemax` return incorrect values whenever
more than one element falls in the support. Sparsemax is defined as the Euclidean
projection onto the probability simplex, so its output must be non-negative and sum to 1
along the reduction axis. It currently does not.

Reproducible on the numpy, jax, tensorflow and torch backends (and by inspection in the
openvino backend, which contains the same line).

### Reproduction

```python
import numpy as np
from keras import ops

x = np.array([[0.0, 0.5, 1.0]], dtype="float32")
print(ops.sparsemax(x))
```

Actual, on every backend:

```
[[0.   0.   0.25]]     # sums to 0.25
```

Expected:

```
[[0.   0.25 0.75]]     # sums to 1.0
```

### Cause

Sparsemax (Martins & Astudillo 2016, Algorithm 1) computes the threshold from the **k-th**
cumulative sum of the descending-sorted logits, where `k` is the size of the support:

```
tau(z) = (cumsum[k] - 1) / k
```

All five backends instead sum *every* cumulative sum that falls inside the support:

```python
logits_cumsum_safe = np.where(support, logits_cumsum, 0.0)   # masks the cumulative sums
tau = (np.sum(logits_cumsum_safe, axis=axis, keepdims=True) - 1) / k
```

For the example above the sorted logits are `[1, 0.5, 0]`, the cumulative sums are
`[1, 1.5, 1.5]`, and the support is the first two entries. The correct threshold is
`(1.5 - 1) / 2 = 0.25`; the code computes `(1 + 1.5 - 1) / 2 = 0.75`.

The same line is present in:

- `keras/src/backend/numpy/nn.py`
- `keras/src/backend/jax/nn.py`
- `keras/src/backend/tensorflow/nn.py`
- `keras/src/backend/torch/nn.py`
- `keras/src/backend/openvino/nn.py`

### Scope

The result is correct only when the support holds exactly one element, because the masked
sum then collapses to `cumsum[0]`. Sampling 2000 random inputs (2–8 logits, normal with
scale 0.3 / 1.0 / 3.0), the threshold is wrong in **1381 of them (69%)**.

The existing tests all happen to fall in the one-element case: `keras/src/ops/nn_test.py`
uses `[-0.5, 0, 1, 2, 3]`, and every case in `keras/src/activations/activations_test.py`
uses `np.linspace(1, 12, ...)`. All produce one-hot output and pass either way.

### Impact

Affects both `keras.ops.sparsemax` and `keras.activations.sparsemax`, so any model using
sparsemax as an activation has been producing incorrect values.

I have a fix ready and will open a PR shortly.

**Comment by hertschuh** [comment]
Fixed by https://github.com/keras-team/keras/pull/23469
